storage/src/mmr/mem.rs

```diff
@@ -122,11 +212,35 @@ impl<const N: usize, H: Hasher<N>> InMemoryMMR<N, H> {
             .map(|(_, peak)| &self.nodes[peak as usize]);
         self.hasher.root_hash(self.nodes.len() as u64, peaks)
     }
+
+    /// Return an inclusion proof for the specified leaf that consists of the size of the MMR and a
+    /// vector of hashes. The proof vector contains: (1) the peak hashes other than the peak of the
+    /// complete subtree containing the leaf, followed by: (2) the nodes in the leaf's complete
+    /// subtree necessary for proving subtree inclusion. Both segments are ordered by decreasing
+    /// height.
+    pub fn proof(&mut self, leaf_pos: u64) -> Proof<N> {
```

Comment:
It would be great if we also had a function that took a range of positions rather than just a single one (for syncing).

Reply:
I think it still makes sense to preserve the single leaf functionality for things like providing a single tx proof.

Reply:
I will benchmark and if the performance difference is negligible I'll just have the single-element case use the range-proof implementation with a single element range.

Reply:
Benchmarks showed only 1-2% performance difference so I have consolidated the implementations.